scripts/repoint-frontend/repoint_frontend.sh

```diff
@@ -0,0 +1,269 @@
+  if [[ $single_step -eq $true && $started -eq $true ]]; then
+    exit 0
+  fi
+
+  # if only listing steps, list step and move on (do not run)
+  if [ $list_steps_only -eq $true ]; then
+    echo "  $((current_step++)): $1"
+    #current_step=$current_step
+    return $false
+  fi
+
+  # if already started, carry on
+  if [ $started -eq $true ]; then
+    spacer
+    step "running: $1"
+    return $true
+  fi
+
+  # increments current step each time function is called (after access)
+  if [ $((current_step++)) -ge $from_step ]; then
+    spacer
+    step "running: $1"
+    started=$true
+    return $true
+  fi
+
+  # step hasn't been reached yet
+  spacer
+  step "skipping: $1"
+  return $false
+}
+
+# doesn't validate the positional args if only listing steps
+if [[ $list_steps_only -eq $false ]]; then
+  # check if a ticket ID was provided
+  if [[ -z "$ticket_id" ]]; then
+    error "missing argument: ticket ID (numeric only)" "usage: $0 <ticket ID: 0000> <shortcode: gith1>"
+  fi
+
+  # check if a ticket shortcode was provided
+  if [[ -z "$shortcode" ]]; then
+    error "missing argument: shortcode" "usage: $0 <ticket ID: 0000> <shortcode: gith1>"
+  fi
+fi
+
+info "initial validation passed"
+
+if check_run_step "remove mTLS (if set)"; then
+  # check if mTLS set on env
+  domain_name="comms-apim.de-$shortcode.communications.national.nhs.uk"
+  domain_status=$(aws apigateway get-domain-name --domain-name $domain_name)
+
+  if echo "$domain_status" | jq -re 'has("mutualTlsAuthentication")' > /dev/null 2>&1; then
+    aws apigateway update-domain-name \
+      --domain-name  $domain_name \
+      --patch-operations op=remove,path=/mutualTlsAuthentication/truststoreUri
+  fi
+fi
+
+if check_run_step "create new branch for proxy modifications"; then
+  current_branch=$(git rev-parse --abbrev-ref HEAD)
+  exit_on_failure "failed to get current git branch"
+
+  repsonse=""
+  info "current branch is: $current_branch"
+  query "do you want to use this branch? (y/n)"
+
+  if [[ "$response" != "y" ]]; then
+    error "checkout the branch you want to work with first"
+  fi
+
+  # create and checkout the new branch using the ticket ID
+  new_branch="CCM-${ticket_id}_REPOINT_DO_NOT_MERGE"
+  git checkout -b "$new_branch"
+  exit_on_failure "failed to checkout branch $new_branch"
+fi
+
+if check_run_step "modify proxy files"; then
+  output=$(python3 scripts/repoint-frontend/repoint_frontend.py $shortcode 2>&1)
+  exit_on_failure "python repoint script failed" "\n $output"
+  info_multiline "$output"
+fi
+
+commit_made=$false
+if check_run_step "stage, commit, and push changes"; then
+  git add proxies
+  exit_on_failure "filed to stage changes (git add)"
+
+  git commit -m "CCM-${ticket_id}: repointed frontend" --no-verify  # verify breaks on my machine, will fix this in due course
+  exit_on_failure "failed to commit changes (git commit)"
+
+  git push --set-upstream origin "$new_branch"
+  exit_on_failure "failed to push changes (git push)"
```

Comment:
Not a bash wizard but I'm unsure how this line would ever hit the exit clause of `exit_on_failure`. I'd presume if line 222 returned a non-zero exit code that this script would just exit out?

Reply:
Bash only automatically exits on non-zero codes if you have `set -e` at the top of the file